src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -618,7 +623,7 @@ public bool VerifySslCertificate
             set
             {
                 if (_isFrozen) { throw new InvalidOperationException("MongoServerSettings is frozen."); }
-                _allowInsecureTls = !value;
+                AllowInsecureTls = !value;
```

Comment:
The fact that you had to use the property instead of the backing field is a bad sign. At the very least add a comment why, because we don't usually do this.

    AllowInsecureTls = !value; // use property instead of private field because setter has additional side effects